app/api/views/briefs.py

```diff
@@ -1382,3 +1420,67 @@ def award_brief_to_seller(brief_id):
         db_object=brief)
 
     return jsonify(work_order=work_order), 200
+
+
+@api.route('/brief/<int:brief_id>/ask-a-question', methods=['POST'])
+@login_required
+@role_required('supplier')
+def supplier_asks_a_question(brief_id):
+    """seller asks a question (role=supplier)
+    ---
+    tags:
+        - brief
+    definitions:
+        BriefAwarded:
+            type: object
+            properties:
+                awardedSupplier:
+                    type: string
+    responses:
+        200:
+            description: Brief awarded successfully.
+            schema:
+                $ref: '#/definitions/BriefAwarded'
+        400:
+            description: Bad request.
+        403:
+            description: Unauthorised to award brief to seller.
+        500:
+            description: Unexpected error.
+    """
+    brief = briefs.get(brief_id)
+    if not brief:
+        not_found('brief {} not found'.format(brief_id))
+
+    data = get_json_from_request()
+    supplier_code = data.get('supplierCode')
```

Comment:
This value in the request and its ensuing check appears to be redundant - the actual reference to the supplier code below when saving the question is coming from the `current_user` (which it should - as we're not currently validating this value from the request data is the same as the requesting user's).

Reply:
fixed